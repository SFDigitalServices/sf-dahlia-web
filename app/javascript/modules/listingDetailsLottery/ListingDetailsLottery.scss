.modal {
  @apply max-w-0;
}

.modal__inner {
  @apply p-0;
}

.modal__close {
  z-index: 100;
}
