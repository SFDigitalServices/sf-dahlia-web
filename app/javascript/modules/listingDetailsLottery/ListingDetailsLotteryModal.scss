.lottery-modal {
    .lottery-results-form .input {
        border-radius: initial
    }
}