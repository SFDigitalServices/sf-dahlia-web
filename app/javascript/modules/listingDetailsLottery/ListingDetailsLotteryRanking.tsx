import React from "react"
import { ExpandableContent, Heading, Icon, t } from "@bloom-housing/ui-components"
import { PREFERENCES } from "../constants"
import { ListingDetailsLotteryResultsRow } from "./ListingDetailsLotteryResultsRow"
import Link from "../../navigation/Link"
import type { RailsLotteryResult } from "../../api/types/rails/listings/RailsLotteryResult"
import { getSfGovUrl, renderMarkup } from "../../util/languageUtil"

interface ListingDetailsLotteryRankingProps {
  lotteryResult: RailsLotteryResult
  listingIsEducator: boolean
}

interface TooltipProps {
  text: string
}

const Tooltip = ({ text }: TooltipProps) => {
  return (
    <div className="bg-blue-100 border-b border-gray-450 flex mb-4 py-4">
      <Icon className="mx-4" size="medium" symbol="info" />
      <p>{text}</p>
    </div>
  )
}
export const ListingDetailsLotteryRanking = ({
  lotteryResult,
  listingIsEducator,
}: ListingDetailsLotteryRankingProps) => {
  const preferenceBuckets = lotteryResult?.lotteryBuckets.filter((bucket) => {
    if (!bucket.preferenceResults[0]) {
      return false
    }
    return bucket.preferenceName !== "generalLottery" && bucket.preferenceResults[0].preferenceRank
  })
  const applicantSelectedForPreference = preferenceBuckets?.length > 0
  const applicantHasCertOfPreference = preferenceBuckets.some(
    (bucket) => bucket.preferenceName === PREFERENCES.certificateOfPreference
  )
  const generalLotteryBucket = lotteryResult?.lotteryBuckets.find(
    (bucket) => bucket.preferenceName === "generalLottery"
  )

  return (
    lotteryResult && (
      <div className="lottery-ranking text-sm">
        <header className="px-8">
          <Heading styleType="underlineWeighted" priority={2}>
            {applicantSelectedForPreference
              ? t("lottery.rankingTitle")
              : t("lottery.rankingTitle.noPreference")}
          </Heading>
          <div className="border-b border-gray-450 pb-4 text-gray-700 translate">
            {renderMarkup(
              `${t("lottery.rankingOrderNote", {
                lotteryRankingVideoUrl: "https://www.youtube.com/watch?v=4ZB35gagUl8",
              })}`
            )}
          </div>
        </header>

        {applicantSelectedForPreference && !applicantHasCertOfPreference && (
          <Tooltip text={t("lottery.rankingPreferenceConsiderationNote")} />
        )}
        {!applicantSelectedForPreference && (
          <Tooltip text={t("lottery.rankingPreferencesConsideredOverGeneralNote")} />
        )}
        {preferenceBuckets?.map((bucket) => (
          <ListingDetailsLotteryResultsRow
            bucket={bucket}
            key={bucket.preferenceName}
            listingIsEducator={listingIsEducator}
          />
        ))}
        {!applicantSelectedForPreference && (
          <ListingDetailsLotteryResultsRow
            bucket={generalLotteryBucket}
            listingIsEducator={listingIsEducator}
          />
        )}
        <div className="px-8">
          <ExpandableContent
            className="pb-4"
            strings={{
              readMore: t("lottery.nextStepsTitle"),
              readLess: t("lottery.nextStepsTitle"),
            }}
          >
            <div className="text-gray-700">
              <p className="mb-2">
                <Link
                  className="text-blue-700"
                  external={true}
                  href={getSfGovUrl("https://sf.gov/after-rental-housing-lottery", 12704)}
                  target="_blank"
                >
                  {t("lottery.nextStepsLearnMore")}
                </Link>
              </p>
              <p className="mb-2">{t("lottery.nextStepsP1")}</p>
              <p className="mb-2">{t("lottery.nextStepsP2")}</p>
              <p className="mb-2">{t("lottery.nextStepsP3")}</p>
            </div>
          </ExpandableContent>
        </div>
      </div>
    )
  )
}
