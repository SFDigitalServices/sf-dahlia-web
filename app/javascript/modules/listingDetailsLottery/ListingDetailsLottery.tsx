import React, { useEffect, useState } from "react"
import {
  AppearanceSizeType,
  AppearanceStyleType,
  Button,
  Heading,
  Modal,
  t,
} from "@bloom-housing/ui-components"
import { RailsListing } from "../listings/SharedHelpers"
import { isLotteryComplete } from "../../util/listingUtil"
import { getLotteryBucketDetails } from "../../api/listingApiService"
import { RailsLotteryResult } from "../../api/types/rails/listings/RailsLotteryResult"
import { ListingDetailsLotteryModal } from "./ListingDetailsLotteryModal"
import { localizedFormat, renderInlineWithInnerHTML } from "../../util/languageUtil"

export interface ListingDetailsLotteryProps {
  listing: RailsListing
}

export const ListingDetailsLottery = ({ listing }: ListingDetailsLotteryProps) => {
  const [lotteryBucketDetails, setLotteryBucketDetails] = useState<RailsLotteryResult>()
  const [isModalOpen, setIsModalOpen] = useState(false)

  useEffect(() => {
    void getLotteryBucketDetails(listing.Id).then((lotteryBucketDetails) => {
      setLotteryBucketDetails(lotteryBucketDetails)
    })
  }, [listing.Id])

  return (
    isLotteryComplete(listing) && (
      <div className="border-b pt-4 text-center">
        <Heading className="mb-4" priority={4}>
          {t("lottery.lotteryResults")}
        </Heading>
        <p className="mb-4 text-sm uppercase">
          {localizedFormat(listing.Lottery_Results_Date, "LL")}
        </p>
        <div className="bg-gray-100 py-4">
          {listing.Lottery_Summary && (
            <div className="mb-3 mx-2 text-gray-700 text-sm">
              {renderInlineWithInnerHTML(listing.Lottery_Summary)}
            </div>
          )}
          <Button
            size={AppearanceSizeType.small}
            styleType={AppearanceStyleType.primary}
            onClick={() => {
              setIsModalOpen(true)
            }}
          >
            {t("listings.lottery.viewLotteryResults")}
          </Button>
        </div>

        {lotteryBucketDetails && (
          <Modal onClose={() => setIsModalOpen(false)} open={isModalOpen} title="">
            <ListingDetailsLotteryModal
              listing={listing}
              lotteryBucketDetails={lotteryBucketDetails}
            />
          </Modal>
        )}
      </div>
    )
  )
}
