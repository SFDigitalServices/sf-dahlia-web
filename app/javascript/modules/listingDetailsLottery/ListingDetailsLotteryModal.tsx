import React, { useState } from "react"
import { Field, Heading, Icon, IconFillColors, Modal, t } from "@bloom-housing/ui-components"
import { useForm } from "react-hook-form"
import { RailsLotteryBucketsDetails } from "../../api/types/rails/listings/RailsLotteryBucketsDetails"
import { RailsListing } from "../listings/SharedHelpers"
import { ListingDetailsLotteryPreferences } from "./ListingDetailsLotteryPreferences"
import "./ListingDetailsLotteryModal.scss"
import { getLotteryResults } from "../../api/listingApiService"
import { RailsLotteryRanking } from "../../api/types/rails/listings/RailsLotteryRanking"
import { ListingDetailsLotteryRanking } from "./ListingDetailsLotteryRanking"
import { ListingDetailsLotteryModalFooter } from "./ListingDetailsLotteryModalFooter"

export enum STATES {
  INITIAL_STATE,
  LOADING,
  API_ERROR,
  INVALID_LOTTERY_NUMBER,
  SEARCH_RESULT_FOUND,
}

interface ListingDetailsLotteryModalProps {
  isOpen: boolean
  listing: RailsListing
  lotteryBucketDetails: RailsLotteryBucketsDetails
  onClose: () => void
}

export const ListingDetailsLotteryModal = ({
  isOpen,
  listing,
  lotteryBucketDetails,
  onClose,
}: ListingDetailsLotteryModalProps) => {
  // eslint-disable-next-line @typescript-eslint/unbound-method
  const { errors, handleSubmit, register } = useForm({ reValidateMode: "onSubmit" })
  const lotteryNumberField = "lotterySearchNumber"
  const [lotterySearchResult, setLotterySearchResult] = useState<RailsLotteryRanking>()

  const [lotteryModalStatus, setLotteryModalStatus] = useState<STATES>(STATES.INITIAL_STATE)

  const onModalClose = () => {
    setLotteryModalStatus(STATES.INITIAL_STATE)
    onClose()
  }

  const onSubmit = (data: { lotterySearchNumber: string }) => {
    const { lotterySearchNumber } = data
    setLotteryModalStatus(STATES.LOADING)

    void getLotteryResults(listing.Id, lotterySearchNumber).then((lotterySearchResults) => {
      if (!lotterySearchResults) {
        // if we didn't get a result, general search api error
        setLotteryModalStatus(STATES.API_ERROR)
        return
      }

      const { lotteryBuckets } = lotterySearchResults
      if (
        // if we got a result but there are no buckets or none of the buckets have pref
        // results, then we don't have a valid lottery number
        !lotteryBuckets ||
        lotteryBuckets.filter((bucket) => bucket.preferenceResults.length > 0).length === 0
      ) {
        setLotteryModalStatus(STATES.INVALID_LOTTERY_NUMBER)
      } else {
        setLotteryModalStatus(STATES.SEARCH_RESULT_FOUND)
        setLotterySearchResult(lotterySearchResults)
      }
    })
  }

  let content: JSX.Element
  switch (lotteryModalStatus) {
    case STATES.LOADING: {
      content = (
        <div className="text-center">
          <Icon size="large" symbol="spinner" />
        </div>
      )
      break
    }
    case STATES.INVALID_LOTTERY_NUMBER: {
      content = (
        <div className="my-6 text-gray-700">
          <p className="mb-2">{t("lottery.lotteryNumberNotFoundP2")}</p>
          <p className="mb-2">{t("lottery.lotteryNumberNotFoundP3")}</p>
        </div>
      )
      break
    }
    case STATES.SEARCH_RESULT_FOUND: {
      content = lotterySearchResult && (
        <ListingDetailsLotteryRanking lotteryRanking={lotterySearchResult} />
      )
      break
    }
    case STATES.INITIAL_STATE: {
      content = <ListingDetailsLotteryPreferences lotteryBucketsDetails={lotteryBucketDetails} />
      break
    }
  }

  const errorMessage =
    (errors[lotteryNumberField]?.type === "required" && t("lottery.lotteryNumberNotValid")) ||
    (lotteryModalStatus === STATES.INVALID_LOTTERY_NUMBER &&
      t("lottery.lotteryNumberNotFoundP1")) ||
    (lotteryModalStatus === STATES.API_ERROR && t("error.lotteryRankingSearch"))

  return (
    <Modal onClose={onModalClose} open={isOpen} title="">
      <div className="lottery-modal w-96">
        <header className="pb-4 text-center">
          <Heading>{t("lottery.lotteryResults")}</Heading>
          <h2 className="font-sans font-semibold text-sm uppercase">{listing.Name}</h2>
        </header>
        <form
          className="bg-gray-100 flex lottery-results-form mb-4 px-6 py-4"
          onSubmit={handleSubmit(onSubmit)}
        >
          <Field
            error={
              errors[lotteryNumberField] ||
              lotteryModalStatus === STATES.API_ERROR ||
              lotteryModalStatus === STATES.INVALID_LOTTERY_NUMBER
            }
            className="w-full"
            errorMessage={
              errorMessage ||
              (errors[lotteryNumberField]?.type === "required" &&
                t("lottery.lotteryNumberNotValid"))
            }
            register={register}
            name={lotteryNumberField}
            placeholder={t("lottery.enterLotteryNumber")}
            type="text"
            validation={{ required: true }}
          />
          <button className="bg-blue-600 h-12 mt-2 p-3">
            <Icon fill={IconFillColors.white} size="medium" symbol="right" />
          </button>
        </form>
        {content}
        <ListingDetailsLotteryModalFooter
          lotteryModalStatus={lotteryModalStatus}
          listing={listing}
        />
      </div>
    </Modal>
  )
}
