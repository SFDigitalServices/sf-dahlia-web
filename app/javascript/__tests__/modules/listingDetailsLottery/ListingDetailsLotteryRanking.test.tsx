import React from "react"
import { render } from "@testing-library/react"
import { ListingDetailsLotteryRanking } from "../../../modules/listingDetailsLottery/ListingDetailsLotteryRanking"
import { lotteryResultRentalTwo } from "../../data/RailsLotteryResult/lottery-result-rental-two"
import { lotteryResultRentalOne } from "../../data/RailsLotteryResult/lottery-result-rental-one"
import { lotteryResultSaleGeneral } from "../../data/RailsLotteryResult/lottery-result-sale-general"

describe("ListingDetailsLotteryRanking", () => {
  it("displays lottery ranking for rental with two results - COP and L/W", () => {
    const { asFragment } = render(
      <ListingDetailsLotteryRanking lotteryResult={lotteryResultRentalTwo} />
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it("displays lottery ranking for rental with one result - L/W", () => {
    const { asFragment } = render(
      <ListingDetailsLotteryRanking lotteryResult={lotteryResultRentalOne} />
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it("displays lottery ranking for sale with general pool only", () => {
    const { asFragment } = render(
      <ListingDetailsLotteryRanking lotteryResult={lotteryResultSaleGeneral} />
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
