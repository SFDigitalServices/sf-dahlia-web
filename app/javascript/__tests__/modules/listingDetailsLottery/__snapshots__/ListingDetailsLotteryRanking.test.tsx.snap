// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingDetailsLotteryRanking displays lottery ranking for educator 1`] = `
<DocumentFragment>
  <div
    class="lottery-ranking text-sm"
  >
    <header
      class="px-8"
    >
      <h2
        class="text__underline-weighted"
      >
        Your preference ranking
      </h2>
      <div
        class="border-b border-gray-450 pb-4 text-gray-700 translate"
      >
        <p>
          Ranking in these lists is considered in the order shown here. 
          <a
            href="https://www.youtube.com/watch?v=4ZB35gagUl8"
            target="_blank"
          >
            Watch a video on how lottery ranking works.
          </a>
        </p>
      </div>
    </header>
    <div
      class="lottery-results-row border-b border-gray-450 flex mb-4 px-8 pb-4"
    >
      <div
        class="bg-blue-500 flex flex-col font-semibold h-full px-4 py-2 text-center text-white"
      >
        <div
          class="rank-title uppercase"
        >
          Rank
        </div>
        <div>
          1
        </div>
      </div>
      <div
        class="mx-4"
      >
        <h3
          class="font-sans font-semibold mb-1 text-sm tracking-wide uppercase"
        >
          Displaced Tenant Housing Preference
        </h3>
        Educators - SFUSD
        <p
          class="mb-1 text-gray-950 text-sm"
        >
          Up to %{units} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 applicants are on this list.
        </p>
      </div>
    </div>
    <div
      class="lottery-results-row border-b border-gray-450 flex mb-4 px-8 pb-4"
    >
      <div
        class="bg-blue-500 flex flex-col font-semibold h-full px-4 py-2 text-center text-white"
      >
        <div
          class="rank-title uppercase"
        >
          Rank
        </div>
        <div>
          1
        </div>
      </div>
      <div
        class="mx-4"
      >
        <h3
          class="font-sans font-semibold mb-1 text-sm tracking-wide uppercase"
        >
          Veteran with Neighborhood Resident Housing Preference
        </h3>
        Educators - SFUSD
        <p
          class="mb-1 text-gray-950 text-sm"
        >
          Up to %{units} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 applicants are on this list.
        </p>
      </div>
    </div>
    <div
      class="lottery-results-row border-b border-gray-450 flex mb-4 px-8 pb-4"
    >
      <div
        class="bg-blue-500 flex flex-col font-semibold h-full px-4 py-2 text-center text-white"
      >
        <div
          class="rank-title uppercase"
        >
          Rank
        </div>
        <div>
          1
        </div>
      </div>
      <div
        class="mx-4"
      >
        <h3
          class="font-sans font-semibold mb-1 text-sm tracking-wide uppercase"
        >
          Neighborhood Resident Housing Preference
        </h3>
        Educators - SFUSD
        <p
          class="mb-1 text-gray-950 text-sm"
        >
          Up to %{units} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 applicants are on this list.
        </p>
      </div>
    </div>
    <div
      class="lottery-results-row border-b border-gray-450 flex mb-4 px-8 pb-4"
    >
      <div
        class="bg-blue-500 flex flex-col font-semibold h-full px-4 py-2 text-center text-white"
      >
        <div
          class="rank-title uppercase"
        >
          Rank
        </div>
        <div>
          1
        </div>
      </div>
      <div
        class="mx-4"
      >
        <h3
          class="font-sans font-semibold mb-1 text-sm tracking-wide uppercase"
        >
          Veteran with Live or Work in San Francisco Preference
        </h3>
        Educators - SFUSD
        <p
          class="mb-1 text-gray-950 text-sm"
        >
          Up to %{units} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          2 applicants are on this list.
        </p>
      </div>
    </div>
    <div
      class="lottery-results-row border-b border-gray-450 flex mb-4 px-8 pb-4"
    >
      <div
        class="bg-blue-500 flex flex-col font-semibold h-full px-4 py-2 text-center text-white"
      >
        <div
          class="rank-title uppercase"
        >
          Rank
        </div>
        <div>
          1
        </div>
      </div>
      <div
        class="mx-4"
      >
        <h3
          class="font-sans font-semibold mb-1 text-sm tracking-wide uppercase"
        >
          Live or Work in San Francisco Preference
        </h3>
        Educators - SFUSD
        <p
          class="mb-1 text-gray-950 text-sm"
        >
          Up to %{units} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          3 applicants are on this list.
        </p>
      </div>
    </div>
    <div
      class="lottery-results-row border-b border-gray-450 flex mb-4 px-8 pb-4"
    >
      <div
        class="bg-blue-500 flex flex-col font-semibold h-full px-4 py-2 text-center text-white"
      >
        <div
          class="rank-title uppercase"
        >
          Rank
        </div>
        <div>
          1
        </div>
      </div>
      <div
        class="mx-4"
      >
        <h3
          class="font-sans font-semibold mb-1 text-sm tracking-wide uppercase"
        >
          Certificate of Preference
        </h3>
        All other employees - SFUSD
        <p
          class="mb-1 text-gray-950 text-sm"
        >
          Up to %{units} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 applicants are on this list.
        </p>
      </div>
    </div>
    <div
      class="lottery-results-row border-b border-gray-450 flex mb-4 px-8 pb-4"
    >
      <div
        class="bg-blue-500 flex flex-col font-semibold h-full px-4 py-2 text-center text-white"
      >
        <div
          class="rank-title uppercase"
        >
          Rank
        </div>
        <div>
          1
        </div>
      </div>
      <div
        class="mx-4"
      >
        <h3
          class="font-sans font-semibold mb-1 text-sm tracking-wide uppercase"
        >
          Veteran with Displaced Tenant Housing Preference
        </h3>
        All other employees - SFUSD
        <p
          class="mb-1 text-gray-950 text-sm"
        >
          Up to %{units} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 applicants are on this list.
        </p>
      </div>
    </div>
    <div
      class="lottery-results-row border-b border-gray-450 flex mb-4 px-8 pb-4"
    >
      <div
        class="bg-blue-500 flex flex-col font-semibold h-full px-4 py-2 text-center text-white"
      >
        <div
          class="rank-title uppercase"
        >
          Rank
        </div>
        <div>
          1
        </div>
      </div>
      <div
        class="mx-4"
      >
        <h3
          class="font-sans font-semibold mb-1 text-sm tracking-wide uppercase"
        >
          Displaced Tenant Housing Preference
        </h3>
        All other employees - SFUSD
        <p
          class="mb-1 text-gray-950 text-sm"
        >
          Up to %{units} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 applicants are on this list.
        </p>
      </div>
    </div>
    <div
      class="lottery-results-row border-b border-gray-450 flex mb-4 px-8 pb-4"
    >
      <div
        class="bg-blue-500 flex flex-col font-semibold h-full px-4 py-2 text-center text-white"
      >
        <div
          class="rank-title uppercase"
        >
          Rank
        </div>
        <div>
          1
        </div>
      </div>
      <div
        class="mx-4"
      >
        <h3
          class="font-sans font-semibold mb-1 text-sm tracking-wide uppercase"
        >
          Neighborhood Resident Housing Preference
        </h3>
        All other employees - SFUSD
        <p
          class="mb-1 text-gray-950 text-sm"
        >
          Up to %{units} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 applicants are on this list.
        </p>
      </div>
    </div>
    <div
      class="lottery-results-row border-b border-gray-450 flex mb-4 px-8 pb-4"
    >
      <div
        class="bg-blue-500 flex flex-col font-semibold h-full px-4 py-2 text-center text-white"
      >
        <div
          class="rank-title uppercase"
        >
          Rank
        </div>
        <div>
          1
        </div>
      </div>
      <div
        class="mx-4"
      >
        <h3
          class="font-sans font-semibold mb-1 text-sm tracking-wide uppercase"
        >
          Veteran with Live or Work in San Francisco Preference
        </h3>
        All other employees - SFUSD
        <p
          class="mb-1 text-gray-950 text-sm"
        >
          Up to %{units} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 applicants are on this list.
        </p>
      </div>
    </div>
    <div
      class="lottery-results-row border-b border-gray-450 flex mb-4 px-8 pb-4"
    >
      <div
        class="bg-blue-500 flex flex-col font-semibold h-full px-4 py-2 text-center text-white"
      >
        <div
          class="rank-title uppercase"
        >
          Rank
        </div>
        <div>
          1
        </div>
      </div>
      <div
        class="mx-4"
      >
        <h3
          class="font-sans font-semibold mb-1 text-sm tracking-wide uppercase"
        >
          Live or Work in San Francisco Preference
        </h3>
        All other employees - SFUSD
        <p
          class="mb-1 text-gray-950 text-sm"
        >
          Up to %{units} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          2 applicants are on this list.
        </p>
      </div>
    </div>
    <div
      class="lottery-results-row border-b border-gray-450 flex mb-4 px-8 pb-4"
    >
      <div
        class="bg-blue-500 flex flex-col font-semibold h-full px-4 py-2 text-center text-white"
      >
        <div
          class="rank-title uppercase"
        >
          Rank
        </div>
        <div>
          1
        </div>
      </div>
      <div
        class="mx-4"
      >
        <h3
          class="font-sans font-semibold mb-1 text-sm tracking-wide uppercase"
        >
          Certificate of Preference (COP)
        </h3>
        General public
        <p
          class="mb-1 text-gray-950 text-sm"
        >
          Up to 1 units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 applicants are on this list.
        </p>
      </div>
    </div>
    <div
      class="lottery-results-row border-b border-gray-450 flex mb-4 px-8 pb-4"
    >
      <div
        class="bg-blue-500 flex flex-col font-semibold h-full px-4 py-2 text-center text-white"
      >
        <div
          class="rank-title uppercase"
        >
          Rank
        </div>
        <div>
          1
        </div>
      </div>
      <div
        class="mx-4"
      >
        <h3
          class="font-sans font-semibold mb-1 text-sm tracking-wide uppercase"
        >
          Displaced Tenant Housing Preference (DTHP)
        </h3>
        General public
        <p
          class="mb-1 text-gray-950 text-sm"
        >
          Up to 1 units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          2 applicants are on this list.
        </p>
      </div>
    </div>
    <div
      class="lottery-results-row border-b border-gray-450 flex mb-4 px-8 pb-4"
    >
      <div
        class="bg-blue-500 flex flex-col font-semibold h-full px-4 py-2 text-center text-white"
      >
        <div
          class="rank-title uppercase"
        >
          Rank
        </div>
        <div>
          1
        </div>
      </div>
      <div
        class="mx-4"
      >
        <h3
          class="font-sans font-semibold mb-1 text-sm tracking-wide uppercase"
        >
          Neighborhood Resident Housing Preference (NRHP)
        </h3>
        General public
        <p
          class="mb-1 text-gray-950 text-sm"
        >
          Up to 1 units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          2 applicants are on this list.
        </p>
      </div>
    </div>
    <div
      class="lottery-results-row border-b border-gray-450 flex mb-4 px-8 pb-4"
    >
      <div
        class="bg-blue-500 flex flex-col font-semibold h-full px-4 py-2 text-center text-white"
      >
        <div
          class="rank-title uppercase"
        >
          Rank
        </div>
        <div>
          1
        </div>
      </div>
      <div
        class="mx-4"
      >
        <h3
          class="font-sans font-semibold mb-1 text-sm tracking-wide uppercase"
        >
          Live or Work in San Francisco Preference
        </h3>
        General public
        <p
          class="mb-1 text-gray-950 text-sm"
        >
          Up to 1 units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          5 applicants are on this list.
        </p>
      </div>
    </div>
    <div
      class="px-8"
    >
      <div
        class="pb-4"
      >
        <button
          aria-expanded="false"
          class="button is-unstyled m-0 no-underline has-toggle"
          type="button"
        >
          What happens next?
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ListingDetailsLotteryRanking displays lottery ranking for rental with one result - L/W 1`] = `
<DocumentFragment>
  <div
    class="lottery-ranking text-sm"
  >
    <header
      class="px-8"
    >
      <h2
        class="text__underline-weighted"
      >
        Your preference ranking
      </h2>
      <div
        class="border-b border-gray-450 pb-4 text-gray-700 translate"
      >
        <p>
          Ranking in these lists is considered in the order shown here. 
          <a
            href="https://www.youtube.com/watch?v=4ZB35gagUl8"
            target="_blank"
          >
            Watch a video on how lottery ranking works.
          </a>
        </p>
      </div>
    </header>
    <div
      class="bg-blue-100 border-b border-gray-450 flex mb-4 py-4"
    >
      <span
        class="ui-icon ui-medium mx-4"
      >
        <svg
          fill="currentColor"
          viewBox="0 0 32 32"
        >
          <path
            d="M14.688 12v-2.688h2.625v2.688h-2.625zM16 26.688c5.875 0 10.688-4.813 10.688-10.688s-4.813-10.688-10.688-10.688-10.688 4.813-10.688 10.688 4.813 10.688 10.688 10.688zM16 2.688c7.375 0 13.313 5.938 13.313 13.313s-5.938 13.313-13.313 13.313-13.313-5.938-13.313-13.313 5.938-13.313 13.313-13.313zM14.688 22.688v-8h2.625v8h-2.625z"
          />
        </svg>
      </span>
      <p>
        Please note, there are other preferences that will be considered for units first.
      </p>
    </div>
    <div
      class="lottery-results-row border-b border-gray-450 flex mb-4 px-8 pb-4"
    >
      <div
        class="bg-blue-500 flex flex-col font-semibold h-full px-4 py-2 text-center text-white"
      >
        <div
          class="rank-title uppercase"
        >
          Rank
        </div>
        <div>
          15
        </div>
      </div>
      <div
        class="mx-4"
      >
        <h3
          class="font-sans font-semibold mb-1 text-sm tracking-wide uppercase"
        >
          Live or Work in San Francisco Preference
        </h3>
        <p
          class="mb-1 text-gray-950 text-sm"
        >
          Up to 1 units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          905 applicants are on this list.
        </p>
      </div>
    </div>
    <div
      class="px-8"
    >
      <div
        class="pb-4"
      >
        <button
          aria-expanded="false"
          class="button is-unstyled m-0 no-underline has-toggle"
          type="button"
        >
          What happens next?
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ListingDetailsLotteryRanking displays lottery ranking for rental with two results - COP and L/W 1`] = `
<DocumentFragment>
  <div
    class="lottery-ranking text-sm"
  >
    <header
      class="px-8"
    >
      <h2
        class="text__underline-weighted"
      >
        Your preference ranking
      </h2>
      <div
        class="border-b border-gray-450 pb-4 text-gray-700 translate"
      >
        <p>
          Ranking in these lists is considered in the order shown here. 
          <a
            href="https://www.youtube.com/watch?v=4ZB35gagUl8"
            target="_blank"
          >
            Watch a video on how lottery ranking works.
          </a>
        </p>
      </div>
    </header>
    <div
      class="lottery-results-row border-b border-gray-450 flex mb-4 px-8 pb-4"
    >
      <div
        class="bg-blue-500 flex flex-col font-semibold h-full px-4 py-2 text-center text-white"
      >
        <div
          class="rank-title uppercase"
        >
          Rank
        </div>
        <div>
          1
        </div>
      </div>
      <div
        class="mx-4"
      >
        <h3
          class="font-sans font-semibold mb-1 text-sm tracking-wide uppercase"
        >
          Certificate of Preference (COP)
        </h3>
        <p
          class="mb-1 text-gray-950 text-sm"
        >
          Up to 1 units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 applicants are on this list.
        </p>
      </div>
    </div>
    <div
      class="lottery-results-row border-b border-gray-450 flex mb-4 px-8 pb-4"
    >
      <div
        class="bg-blue-500 flex flex-col font-semibold h-full px-4 py-2 text-center text-white"
      >
        <div
          class="rank-title uppercase"
        >
          Rank
        </div>
        <div>
          374
        </div>
      </div>
      <div
        class="mx-4"
      >
        <h3
          class="font-sans font-semibold mb-1 text-sm tracking-wide uppercase"
        >
          Live or Work in San Francisco Preference
        </h3>
        <p
          class="mb-1 text-gray-950 text-sm"
        >
          Up to 1 units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          905 applicants are on this list.
        </p>
      </div>
    </div>
    <div
      class="px-8"
    >
      <div
        class="pb-4"
      >
        <button
          aria-expanded="false"
          class="button is-unstyled m-0 no-underline has-toggle"
          type="button"
        >
          What happens next?
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ListingDetailsLotteryRanking displays lottery ranking for sale with general pool only 1`] = `
<DocumentFragment>
  <div
    class="lottery-ranking text-sm"
  >
    <header
      class="px-8"
    >
      <h2
        class="text__underline-weighted"
      >
        Your lottery ranking
      </h2>
      <div
        class="border-b border-gray-450 pb-4 text-gray-700 translate"
      >
        <p>
          Ranking in these lists is considered in the order shown here. 
          <a
            href="https://www.youtube.com/watch?v=4ZB35gagUl8"
            target="_blank"
          >
            Watch a video on how lottery ranking works.
          </a>
        </p>
      </div>
    </header>
    <div
      class="bg-blue-100 border-b border-gray-450 flex mb-4 py-4"
    >
      <span
        class="ui-icon ui-medium mx-4"
      >
        <svg
          fill="currentColor"
          viewBox="0 0 32 32"
        >
          <path
            d="M14.688 12v-2.688h2.625v2.688h-2.625zM16 26.688c5.875 0 10.688-4.813 10.688-10.688s-4.813-10.688-10.688-10.688-10.688 4.813-10.688 10.688 4.813 10.688 10.688 10.688zM16 2.688c7.375 0 13.313 5.938 13.313 13.313s-5.938 13.313-13.313 13.313-13.313-5.938-13.313-13.313 5.938-13.313 13.313-13.313zM14.688 22.688v-8h2.625v8h-2.625z"
          />
        </svg>
      </span>
      <p>
        Please note, preference holders will be considered for all units first.
      </p>
    </div>
    <div
      class="lottery-results-row border-b border-gray-450 flex mb-4 px-8 pb-4"
    >
      <div
        class="bg-blue-500 flex flex-col font-semibold h-full px-4 py-2 text-center text-white"
      >
        <div
          class="rank-title uppercase"
        >
          Rank
        </div>
        <div>
          1
        </div>
      </div>
      <div
        class="mx-4"
      >
        <h3
          class="font-sans font-semibold mb-1 text-sm tracking-wide uppercase"
        >
          General Pool
        </h3>
        <p
          class="mb-1 text-gray-950 text-sm"
        >
          Any remaining units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          2 applicants are on this list.
        </p>
      </div>
    </div>
    <div
      class="px-8"
    >
      <div
        class="pb-4"
      >
        <button
          aria-expanded="false"
          class="button is-unstyled m-0 no-underline has-toggle"
          type="button"
        >
          What happens next?
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
